clang/lib/Parse/ParseOpenACC.cpp

```diff
@@ -262,12 +291,52 @@ OpenACCDirectiveKind ParseOpenACCDirectiveKind(Parser &P) {
   return DirKind;
 }
 
+bool ParseOpenACCClause(Parser &P) {
```

Comment:
Can you add a description of the grammar?

Reply:
So the individual clauses each have their own 'grammar'.  I DID comment the 'clause-list' below best I could, but there is no  clause 'grammar' in the standard.  The one's ive mentioned are the ones where it is all a single identifier-token (or 'auto' as you've seen :/).

I'll spend a few moments typing something up to make this more clear however.

Reply:
Thanks